From akpm@linux-foundation.org  Mon Nov 26 15:01:46 2007
From: Michal Schmidt <mschmidt@redhat.com>
Date: Wed, 21 Nov 2007 15:17:53 -0800
Subject: PCI: use pci=bfsort for HP DL385 G2, DL585 G2
To: tglx@linutronix.de
Cc: mingo@elte.hu, akpm@linux-foundation.org, mschmidt@redhat.com, Matt_Domsch@dell.com, ak@suse.de, andy@greyhouse.net, greg@kroah.com, john.cagle@hp.com
Message-ID: <200711212317.lALNHr3v014421@imap1.linux-foundation.org>


From: Michal Schmidt <mschmidt@redhat.com>

HP ProLiant systems DL385 G2 and DL585 G2 need pci=bfsort to enumerate PCI
devices in the expected order.

Matt sayeth:

  biosdevname is a userspace app I wrote to help solve this so we don't need
  to patch the kernel for future systems.  It's not integrated into any
  distributions properly yet, but is included in openSUSE 10.3 and Fedora 8
  for people who want to download and install it there.  It acts as a udev
  helper.

  For the time being, patching the kernel is necessary.  I really hope
  biosdevname eliminates that need in future distributions.

  http://linux.dell.com/biosdevname/

Signed-off-by: Michal Schmidt <mschmidt@redhat.com>
Cc: <john.cagle@hp.com>
Cc: <tglx@linutronix.de>
Cc: Matt Domsch <Matt_Domsch@dell.com>
Acked-by: Andy Gospodarek <andy@greyhouse.net>
Cc: Andi Kleen <ak@suse.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/x86/pci/common.c |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- a/arch/x86/pci/common.c
+++ b/arch/x86/pci/common.c
@@ -315,6 +315,22 @@ static struct dmi_system_id __devinitdat
 		},
 	},
 #endif
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant DL385 G2",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant DL385 G2"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant DL585 G2",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant DL585 G2"),
+		},
+	},
 	{}
 };
 
